### ISSUE TITLE
Windows 8.1 / Portable 2.6.2 / ln: failed to create hard link

### ISSUE THREAD

**Issue by albertosantini** [git-for-windows/git#506]
After downloading `PortableGit-2.6.2-64-bit.7z.exe`, I executed it and extracted the files in `C:\My\Programs\Git5`.

When I started `git-bash.exe`, I get the following messages before the prompt.

```
ln: failed to create hard link 'C:\\My\\Programs\\Git5\\mingw64/libexec/git-core/libtre-5.dll': File exists
cp: 'C:\\My\\Programs\\Git5\\mingw64/bin/libtre-5.dll' and 'C:\\My\\Programs\\Git5\\mingw64/libexec/git-core/libtre-5.dll' are the same file
ERROR: could not link C:\My\Programs\Git5\mingw64/bin/libtre-5.dll C:\My\Programs\Git5\mingw64/libexec/git-core
ln: failed to create hard link 'C:\\My\\Programs\\Git5\\mingw64/libexec/git-core/libvtv_stubs-0.dll': File exists
cp: 'C:\\My\\Programs\\Git5\\mingw64/bin/libvtv_stubs-0.dll' and 'C:\\My\\Programs\\Git5\\mingw64/libexec/git-core/libvtv_stubs-0.dll' are the same file
ERROR: could not link C:\My\Programs\Git5\mingw64/bin/libvtv_stubs-0.dll C:\My\Programs\Git5\mingw64/libexec/git-core
ln: failed to create hard link 'C:\\My\\Programs\\Git5\\mingw64/libexec/git-core/ssleay32.dll': File exists
cp: 'C:\\My\\Programs\\Git5\\mingw64/bin/ssleay32.dll' and 'C:\\My\\Programs\\Git5\\mingw64/libexec/git-core/ssleay32.dll' are the same file
ERROR: could not link C:\My\Programs\Git5\mingw64/bin/ssleay32.dll C:\My\Programs\Git5\mingw64/libexec/git-core
ln: failed to create hard link 'C:\\My\\Programs\\Git5\\mingw64/libexec/git-core/tk86.dll': File exists
cp: 'C:\\My\\Programs\\Git5\\mingw64/bin/tk86.dll' and 'C:\\My\\Programs\\Git5\\mingw64/libexec/git-core/tk86.dll' are the same file
ERROR: could not link C:\My\Programs\Git5\mingw64/bin/tk86.dll C:\My\Programs\Git5\mingw64/libexec/git-core
```

It happens only once and the list of the files involved changes.
I repeated a few times the task, installing in `Git1`, `Git2`, and so on, getting always that list of messages.

(Windows 8.1 64bit and PortableGit-2.6.2-64-bit)

**Comment by dscho** [comment]
Have you run `post-install.bat` as recommended in `README.portable`?

**Comment by dscho** [comment]
Hmm. The [responsible script](https://github.com/git-for-windows/build-extra/blob/master/portable/root/etc/post-install/13-copy-dlls.post) should

- run only once (when `post-install.bat` is executed, which is done automatically if you use the self-extractor), then be deleted, and
- test whether the `.dll` files exist before trying to hard link or copy.

Could you unpack that archive again and see what the status is *before* running a `git-bash.exe` instance? In particular, I am interested whether `etc/post-install/` is empty and whether there are already a couple of `.dll` files in `mingw64/libexec/git-core/`.

If this analysis does not shed any light into the matter, I would ask you to insert the command `set -x` into `etc/profile` (at the top) before running the first bash session, and then inspect the output to figure out *where* the `ln` and `cp` commands were called.

**Comment by dscho** [comment]
(To be clear: I would perform this investigation myself if I had a way to reproduce the issue. But I do not.)

**Comment by albertosantini** [comment]
After ten tries, I could not reproduce it (because I was too slow).
At least, I understood the issue (because I was too fast).

After clicking on `PortableGit-2.6.2-64-bit.7z.exe`, the extraction starts; then when it ends, it executed automatically `post-install.bat`; usually it ends in a few seconds.

If I start `git-bash.exe` in the meanwhile, before post process is ended, I get the issue.

Maybe a few hours ago, my box was under load and post install process get almost a 30 secs to complete; so I noticed those messages. When I tried to reply the issue, I extracted the file and immediately executed the bash shell (creating the issue).

Thanks again for your time.
I close the issue.

**Comment by dscho** [comment]
> If I start git-bash.exe in the meanwhile, before post process is ended, I get the issue.

Ah, that explains it! Well, I guess I never thought that anybody would launch a program before it is fully installed ;-)

**Comment by albertosantini** [comment]
Well... no busy (spin or in progress) cursor here. :)
After the extraction, the installation seems ended. 

For me it is a `won't fix`, but maybe adding in `README.portable` a sentence like the following one may be feasible:

*NOTE*: [...cut...] Git is ready when there is not `post-install.bat` script in the target directory. Git will not run correctly otherwise.
